File: Client/src/app/repozytorium-pamieciowe.service.ts
import { Injectable } from '@angular/core';
import { GetDataInterface } from './interfaces/get-data.interface';
import { FormSubmitInterface } from './interfaces/form-submit.interface';
import { Observable, of } from 'rxjs';
import { GraClass } from './classes/gra.class';

@Injectable()
export class RepozytoriumPamiecioweService implements GetDataInterface, FormSubmitInterface {
  private idGenerator = 1;
  private readonly data: GraClass[] = [
    new GraClass(this.idGenerator++, "Wiedźmin 3", 129.99, new Date(2015, 4, 19)),
    new GraClass(this.idGenerator++, "Cyberpunk 2077", 199.99, new Date(2020, 11, 10)),
    new GraClass(this.idGenerator++, "Minecraft", 89.99, new Date(2011, 10, 18)),
    new GraClass(this.idGenerator++, "Grand Theft Auto V", 149.99, new Date(2013, 8, 17)),
    new GraClass(this.idGenerator++, "The Elder Scrolls V: Skyrim", 119.99, new Date(2011, 10, 11))
  ]

  Post(nazwa: string, cena: number, data: Date): Observable<boolean> {
    const newObj = new GraClass(this.idGenerator++, nazwa, cena, data);
    this.data.push(newObj);
    return of(true);
  }

  Put(id: number, nazwa: string, cena: number, data: Date): Observable<boolean> {
    const obj = this.data.find(x => x.id === id);
    if(obj) {
      obj.tytul = nazwa;
      obj.cena = cena;
      obj.dataPremiery = data;
      return of(true);
    }

    return of(false);
  }

  Get(): Observable<GraClass[]> {
    const kopia = this.data.map(x => new GraClass(x.id, x.tytul, x.cena, x.dataPremiery));
    return of(kopia);
  }

  GetByID(id: number): Observable<GraClass> {
    const obj = this.data.find(x => x.id === id);
    if(obj) {
      const kopia = new GraClass(obj.id, obj.tytul, obj.cena, obj.dataPremiery);
      return of(kopia);
    }

    throw new Error("Nie znaleziono obiektu.");
  }
}

File: WebAPI/WebAPI/Services/RepozytoriumPamiecioweService.cs
using WebAPI.Models;

namespace WebAPI.Services
{
    public class RepozytoriumPamiecioweService
    {
        private int _idGenerator = 1;
        private readonly List<Gra> _data = new();

        public RepozytoriumPamiecioweService()
        {
            _data.Add(new Gra { Id = _idGenerator++, Tytul = "Wiedźmin 3", Cena = 129.99m, DataPremiery = new DateTime(2015, 5, 19) });
            _data.Add(new Gra { Id = _idGenerator++, Tytul = "Cyberpunk 2077", Cena = 199.99m, DataPremiery = new DateTime(2020, 12, 10) });
            _data.Add(new Gra { Id = _idGenerator++, Tytul = "Minecraft", Cena = 89.99m, DataPremiery = new DateTime(2011, 11, 18) });
            _data.Add(new Gra { Id = _idGenerator++, Tytul = "Grand Theft Auto V", Cena = 149.99m, DataPremiery = new DateTime(2013, 9, 17) });
            _data.Add(new Gra { Id = _idGenerator++, Tytul = "The Elder Scrolls V: Skyrim", Cena = 119.99m, DataPremiery = new DateTime(2011, 11, 11) });
        }

        public List<Gra> Get() => _data.Select(x => new Gra
        {
            Id = x.Id,
            Tytul = x.Tytul,
            Cena = x.Cena,
            DataPremiery = x.DataPremiery
        }).ToList();

        public Gra? GetById(int id) => _data.FirstOrDefault(x => x.Id == id);

        public bool Post(string tytul, decimal cena, DateTime dataPremiery)
        {
            _data.Add(new Gra { Id = _idGenerator++, Tytul = tytul, Cena = cena, DataPremiery = dataPremiery });
            return true;
        }

        public bool Put(int id, string tytul, decimal cena, DateTime dataPremiery)
        {
            var obj = _data.FirstOrDefault(x => x.Id == id);
            if (obj == null) return false;
            obj.Tytul = tytul;
            obj.Cena = cena;
            obj.DataPremiery = dataPremiery;
            return true;
        }

        public bool Delete(int id)
        {
            var obj = _data.FirstOrDefault(x => x.Id == id);
            if (obj == null) return false;
            _data.Remove(obj);
            return true;
        }
    }
}
